From c9a035aeb8914f331aa809a44365670ef63d80d4 Mon Sep 17 00:00:00 2001
From: Alexis Montoison <alexis.montoison@polymtl.ca>
Date: Tue, 6 Jun 2023 23:48:25 -0400
Subject: [PATCH 1/2] Linear solvers such as MUMPS should prefered over SPRAL

---
 src/Algorithm/IpAlgBuilder.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/Algorithm/IpAlgBuilder.cpp b/src/Algorithm/IpAlgBuilder.cpp
index 8c622c17f..83d66cd50 100644
--- a/src/Algorithm/IpAlgBuilder.cpp
+++ b/src/Algorithm/IpAlgBuilder.cpp
@@ -225,10 +225,6 @@ void AlgorithmBuilder::RegisterOptions(
    {
       defaultsolver = "pardiso";
    }
-   else if( availablesolverslinked & IPOPTLINEARSOLVER_SPRAL )
-   {
-      defaultsolver = "spral";
-   }
    else if( availablesolverslinked & IPOPTLINEARSOLVER_WSMP )
    {
       defaultsolver = "wsmp";
@@ -241,6 +237,10 @@ void AlgorithmBuilder::RegisterOptions(
    {
       defaultsolver = "pardisomkl";
    }
+   else if( availablesolverslinked & IPOPTLINEARSOLVER_SPRAL )
+   {
+      defaultsolver = "spral";
+   }
    else if( availablesolverslinked & IPOPTLINEARSOLVER_MA77 )
    {
       defaultsolver = "ma77";

From cd54547ae62a404eea80d9fa6ec6ddef5c805ad1 Mon Sep 17 00:00:00 2001
From: Stefan Vigerske <svigerske@users.noreply.github.com>
Date: Wed, 7 Jun 2023 09:27:32 +0200
Subject: [PATCH 2/2] add changelog entry

---
 ChangeLog.md | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ChangeLog.md b/ChangeLog.md
index b834829c7..c82bde275 100644
--- a/ChangeLog.md
+++ b/ChangeLog.md
@@ -7,6 +7,10 @@ More detailed information about incremental changes can be found in the
 
 ## 3.14
 
+### 3.14.13 (2023-xx-yy)
+
+- Reduced priority for making Spral the default value for option linear_solver [#677].
+
 ### 3.14.12 (2023-04-05)
 
 - Fix that a source file was installed and install more header files.
