--- a/src/Algorithm/LinearSolvers/IpMumpsSolverInterface.cpp	2017-04-26 01:45:44.334073203 +0200
+++ a/src/Algorithm/LinearSolvers/IpMumpsSolverInterface.cpp	2017-04-26 01:52:13.063595789 +0200
@@ -16,11 +16,7 @@
 // The first header to include is the one for MPI.  
 // In newer ThirdParty/Mumps, mpi.h is renamed to mumps_mpi.h.
 // We get informed about this by having COIN_USE_MUMPS_MPI_H defined.
-#ifdef COIN_USE_MUMPS_MPI_H
 #include "mumps_mpi.h"
-#else
-#include "mpi.h"
-#endif
 
 #include "IpMumpsSolverInterface.hpp"
 
